releasenotes/notes/singletons-83782de8bd062cbc.yaml

```diff
@@ -0,0 +1,122 @@
+        * :class:`~.SXdgGate`
+        * :class:`~.TGate`
+        * :class:`~.TdgGate`
+        * :class:`~.XGate`
+        * :class:`~.RCCXGate`
+        * :class:`~.RC3XGate`
+        * :class:`~.YGate`
+        * :class:`~.ZGate`
+
+    This means that unless a ``label``, ``condition``, ``duration``, or ``unit``
+    are set on the instance at creation time they will all share a single global
+    instance whenever a new gate object is created. This results in large reduction
+    in the memory overhead for > 1 object of these types and significantly faster
+    object construction time.
+  - |
+    Added a new method :meth`.Instruction.to_mutable` and attribute
+    :attr:`.Instruction.mutable` which is used to get a mutable copy and check whether
+    an :class:`~.circuit.Instruction` object is mutable. With the introduction
+    of :class:`~.SingletonGate` these methods can be used to have a unified interface
+    to deal with the mutablitiy of instruction objects.
+upgrade:
+  - |
+    The following standard library gates:
+
+        * :class:`~.DCXGate`
+        * :class:`~.ECRGate`
+        * :class:`~.HGate`
+        * :class:`~.IGate`
+        * :class:`~.iSwapGate`
+        * :class:`~.SGate`
+        * :class:`~.SdgGate`
+        * :class:`~.SwapGate`
+        * :class:`~.SXGate`
+        * :class:`~.SXdgGate`
+        * :class:`~.TGate`
+        * :class:`~.TdgGate`
+        * :class:`~.XGate`
+        * :class:`~.RCCXGate`
+        * :class:`~.RC3XGate`
+        * :class:`~.YGate`
+        * :class:`~.ZGate`
+
+    no longer are able to set :attr:`~.Gate.label`, :attr:`~.Gate.condition`,
+    :attr:`~.Gate.duration`, or :attr:`~.Gate.unit` after instantiating an object
+    anymore. You will now only be able to set these attributes as arguments
+    when creating a new object or in the case of :attr:`~.Gate.condtion` through
+    the use :meth:`~.Gate.c_if`. Alternatively you can use :meth:`~.Gate.to_mutable`
+    to get a mutable copy of the instruction and then use the setter on that copy
+    instead of the original object. This change was necssary as part of converting
+    these classes to be :class:`~.SingletonGate` types which greatly reduces the
+    memory footprint of repeated instances of these gates.
+  - |
+    For anything that interacts with :class:`~.Gate`, :class:`~.Operation`,
+    or :class:`~.circuit.Instruction` objects or works with these as part of a
+    :class:`~.QuantumCircuit` or :class:`~.DAGCircuit` classes it is important
+    to note that the use of shared references for instances is much more common
+    now. Previously, it was possible to reuse and share an instance of a
+    a circuit operation it wasn't very commonly used and a copy would generate
+    a unique instance. This has changed starting in this release because of
+    :class:`~.SingletonGate` being made available (and a large number of standard
+    library gates now built off of it). If you're use of these objects is assuming
```

Comment:
```suggestion
    library gates now built off of it). If your usage of these objects is assuming
```

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/10314/commits/9c9bc0999f892bcf558fff52629bc89471b3f414